saleor/graphql/account/mutations/permission_group.py

```diff
@@ -107,118 +214,102 @@ class Meta:
+            group.permissions.add(*cleaned_input["add_permissions"])
+        if "remove_perissions" in cleaned_input:
+            remove_permissions = get_permissions(cleaned_input["remove_perissions"])
+            group.permissions.remove(*remove_permissions)
 
-        user_pks = [
-            from_global_id_strict_type(user_id, only_type=User, field="id")
-            for user_id in user_ids
-        ]
+    @classmethod
+    def clean_input(
+        cls, info, instance, data,
+    ):
+        cleaned_input = super().clean_input(info, instance, data)
+        errors = defaultdict(list)
+        for field in ["users", "permissions"]:
+            cls.check_for_duplicates(errors, cleaned_input, field)
 
-        return user_pks
+        cls.clean_permissions(
+            info, errors, "add_permissions", cleaned_input, "permissions"
+        )
+        cls.clean_users(errors, "add_users", cleaned_input, "users")
+
+        if errors:
+            raise ValidationError(errors)
 
-    @classmethod
-    def clean_input(cls, info, instance, data, input_cls=None):
-        cleaned_input = super().clean_input(info, instance, data, input_cls=input_cls)
-        user_ids: List[str] = cleaned_input["users"]
-        if not user_ids:
-            raise ValidationError(
-                {
-                    "users": ValidationError(
-                        "You must provide at least one staff user.",
-                        code=AccountErrorCode.REQUIRED.value,
-                    )
-                }
-            )
         return cleaned_input
 
-    @staticmethod
-    def check_if_users_are_staff(user_pks: List[int]):
-        non_staff_users = account_models.User.objects.filter(pk__in=user_pks).filter(
-            is_staff=False
-        )
-        if non_staff_users:
-            raise ValidationError(
-                {
-                    "users": ValidationError(
-                        "Some of users aren't staff members.",
-                        code=AccountErrorCode.ASSIGN_NON_STAFF_MEMBER.value,
-                    )
-                }
+    @classmethod
+    def check_for_duplicates(
+        cls,
+        errors: Dict[Optional[str], List[ValidationError]],
+        cleaned_input: dict,
+        field: str,
+    ):
+        """Check if some of the items are in add and remove list at the same time.
+
+        Raise error if some of items are duplicated.
+        """
+        add_field = f"add_{field}"
+        remove_field = f"remove_{field}"
+        if add_field in cleaned_input and remove_field in cleaned_input:
+            # get items which are in both list
+            common_items = set(cleaned_input[add_field]) & set(
+                cleaned_input[remove_field]
             )
+            if common_items:
+                if field == "permission":
+                    # prepare permissions enum list for error
+                    values = [PermissionEnum.get(perm) for perm in common_items]
```

Comment:
Do we have to do this transformation? Aren't `common_items` already list of permission names that were provided in the input and could be returned directly to the error params?

Reply:
Yeah, I missed it.